salt/master.py

```diff
@@ -1957,6 +1959,15 @@ def publish(self, clear_load):
         payload = self._prep_pub(minions, jid, clear_load, extra, missing)
 
         # Send it!
+        if self.opts[u'enable_ssh_minions'] is True and isinstance(clear_load[u'tgt'], six.string_types):
```

Comment:
This method (`publish`) has so much bloat, I think it would be best to split this out into a private method.